.github/workflows/roles-readme.yml

```
@@ -0,0 +1,55 @@
+---
+name: create roles readme
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - master
+
+jobs:
+  readme:
+    name: create roles readme
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        roles:
+          - "mysql_hardening"
+          - "nginx_hardening"
+          - "os_hardening"
+          - "ssh_hardening"
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          repository: telekom-mms/Automated-Ansible-Role-Documentation
+          path: aar-doc/
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Install poetry
+        run: pip3 install poetry
+
+      - name: Install dependencies
+        run: poetry install
+        working-directory: aar-doc/
+
+      - name: Run Python Script
+        run: poetry run python3 aar_doc/cli.py ../roles/${{ matrix.roles }} markdown
+        working-directory: aar-doc/
```

aar_doc is available in the Pypi package registry, so there's no need to download it and install it with poetry.

```suggestion
      - name: Set up Python
        uses: actions/setup-python@v4
        with:
          python-version: 3.11

      - name: Install aar_doc
        run: pip3 install aar_doc
        
      - name: Run aar_doc
        run:  aar_doc ../roles/${{ matrix.roles }} markdown
        working-directory: aar-doc/
```